src/memory.rs

```diff
@@ -1,17 +1,26 @@
-use crate::constants::MEMORY_MAX;
+use crate::utils::{handle_keyboard, MemoryMappedReg};
+
+pub const MEMORY_MAX: u16 = u16::MAX;
 
 pub struct Memory {
     memory: [u16; MEMORY_MAX as usize + 1],
 }
 
 impl Memory {
-    pub fn read(&self, addr: u16) -> u16 {
+    pub fn read(&mut self, addr: u16) -> u16 {
+        if addr == MemoryMappedReg::Kbsr as u16 {
+            handle_keyboard(self);
+        }
         self.memory[addr as usize]
     }
 
     pub fn write(&mut self, addr: u16, data: u16) {
         self.memory[addr as usize] = data;
     }
+
+    pub fn size() -> u16 {
+        MEMORY_MAX
+    }
```

Comment:
This function is never used.

Reply:
True, it had a purpose at some point I think